### ISSUE TITLE
POST Comes In As GET

### ISSUE THREAD

**Issue by JimTheMan** [awslabs/aws-serverless-express#145]
Hi, whenever I send an ajax POST request like this:

```
$.ajax({
			url: 		options.url,
			dataType: 	'json',
			crossDomain: true,
			contentType: "application/json",
			type:		"POST",
			data:		options.params,
			success: 	function(data){
				// Apply the ajax-after plugins: Modifying the ajax response
				_.each(scope.plugins['ajax-after'], function(callback) {
					data	= callback(data, options.params);
				});
				options.callback(data);
				console.log('@@@, ', data)
			},
			error: function(jqXHR, data, errorThrown) {
				
			}
		});
```

The request is handled by my lambda function as a GET request.

```
const awsServerlessExpressMiddleware = require('aws-serverless-express/middleware')
const userEmbed = require('./src/user-embed')
const bodyParser = require('body-parser')
const express = require('express')
const app = express()
app.use(express.urlencoded({extended: true}))
app.use(bodyParser.json())
app.use(awsServerlessExpressMiddleware.eventContext())

// Allows supertest e2e tests to override the apiGateway event object.
app.inputObjectOverride = { uuids: '' }

app.get('/', function(req, res) {

	let args = app.inputObjectOverride

  if (req.apiGateway)
    args = req.apiGateway.event.queryStringParameters || req.apiGateway.event

    console.log('get args: ', args)

	return userEmbed.handleIncomingUuidsOrUuid(args.uuids).then(result => {

	  console.log('Returning Get result! ', result);
		res.send(result)
	}, error => {
		res.send(error)
    })
})

var urlencodedParser = bodyParser.urlencoded({ extended: false })

app.post('/', function(req, res) {
  let args = app.inputObjectOverride

  if (req.apiGateway)
    args = req.apiGateway.event.queryStringParameters || req.apiGateway.event

  console.log('post args: ', args)

  return userEmbed.handleIncomingUuidsOrUuid(args.uuids).then(result => {

    console.log('Returning Post result! ', result)
    res.send(result)
  }, error => {
    res.send(error)
  })
})

module.exports = app
```

The log shows "get args:" and "Returning Get result!". Is this intentional? Why is it being handled by `app.get` instead of `app.post`?

Thanks!

**Comment by brettstack** [comment]
Have you confirmed a POST request is being sent from the client and not a GET request?

**Comment by JimTheMan** [comment]
Hey thanks! Yes, it looks like a post request being sent from the client: 

![fleetwit_-_brain_games_for_cash_](https://user-images.githubusercontent.com/5354163/40067341-f76951e6-5833-11e8-88c2-5f9260c16a23.png)

**Comment by JimTheMan** [comment]
Btw in the api gateway method integration settings I have "Use Lambda Proxy integration" unchecked. No headers or body mapping template.

**Comment by JimTheMan** [comment]
Thanks, but I want to return custom headers for CORS and stuff like that. How can you do that using proxy integration?

**Comment by JimTheMan** [comment]
I just created a new ANY method with Lambda proxy integration checked, but it didn't help. I guess I need to check proxy integration for POST as well?

**Comment by JimTheMan** [comment]
I just checked the box for POST and deployed again. It is now being handled by app.post. yay!

Not sure why my browser it letting go through though since my POST response isn't returning any "access-control-allow-origin" header.

**Comment by brettstack** [comment]
You can return any headers you need from your Express route. Also for most scenarios I recommend using the special ANY (which will simply pass through the original request method) method instead of individual methods

**Comment by JimTheMan** [comment]
But when I delete the GET and POST resources in API gateway the query parameters don't get mapped correctly from the POST body.

**Comment by brettstack** [comment]
Query parameters don't get mapped from the POST body? Query parameters are different than body and show up under a different property.

**Comment by brettstack** [comment]
That's only if it's application/x-www-form-urlencoded

**Comment by JimTheMan** [comment]
Ok, thanks Brett. I was able to get it working by following the project on this repo's example folder, but I already had those libraries in my project so I'm still confused as to my my old version didn't work... is the order of the "app.use" statements very important?

Also, I had to remove the "compression" lib. When I had this in there the code would get all the way down to "res.send", but it wouldn't actually give me a response. Any idea why this is would be happening? The data in trying to return in an array containing one "medium" sized object.
